pygazebo/pygazebo.cc

```diff
@@ -252,18 +254,59 @@ class Agent : public Model {
             reinterpret_cast<const uint8_t*>(sensor->ImageData())));
   }
 
-  bool TakeAction(const std::map<std::string, double>& forces) {
+  bool TakeAction(const std::map<std::string, double>& controls) {
     auto controller = model_->GetJointController();
-    for (const auto& name2force : forces) {
-      bool ret = controller->SetForce(name2force.first, name2force.second);
+    for (const auto& name2control : controls) {
+      bool ret;
+      int control_type = joints_control_type_[name2control.first];
```

Comment:
need to default the control_type to control_type_force_ because the user may not always call SetPIDController()

Reply:
It's safe here. Value in std::map<std::string, int> is 0 (Forces) by default, and it will be created automaticly at the line control_type = joints_control_type_[jointname] if that jointname is not existing. Setting defaut explicitly when agent is creating needs to repeat the operation GetJointNames() since we don't know the joint list, so we can save some initializing codes.

Std::map overload operator "[ ]", it will find the key. but if the key is not existing, then create one with default value, int() here so it's 0. (So we cannot use joint_control_type.find here).

Reply:
You are right. I confused it with python map.